server/routers/auth.js

```diff
@@ -0,0 +1,96 @@
+const express = require('express');
+const path = require('path');
+require('dotenv').config({ path: path.join(__dirname, '../.env') });
+const { Types } = require('mongoose');
+const { User, Cart } = require('../models');
+const { signToken, verifyToken } = require('../jwt');
+
+const auth = express.Router();
+const { ObjectId } = Types;
+
+auth.post('/register', (request, response) => {
+  if (!request.body) {
+    return response.sendStatus(400);
+  }
+
+  const { email, password } = request.body;
+  const user = new User({
+    email,
+    password,
+  });
+
+  return User.findOne({})
+    .then((result) => {
+      // first registered user become admin
+      user.isAdmin = !result;
+    })
+    .then(() =>
+      User.findOne({ email }).then((result) => {
+        if (result) {
+          // users with same e-mails not allowed
+          return response.sendStatus(409);
+        }
+        return user.save().then((newUser) => {
+          const { _id, isAdmin, isActive } = newUser;
+          const token = signToken({ _id, isAdmin, isActive });
+          response.clearCookie('cartId', { secure: false, maxAge: 0 });
+          response.cookie('token', token, { secure: false, maxAge: 3600 * 24 });
+          response.json(email);
+        });
+      })
+    );
+});
+
+auth.post('/login', (request, response) => {
+  if (!request.body) {
+    return response.sendStatus(400);
+  }
+  const { email, password } = request.body;
+
+  return User.findOneAndUpdate({ email, password }, { isActive: true }, { new: true }).then(
```

Comment:
@IgorChaikin Basically `isActive` shouldn't be updated on login, it allows admin to ban some users, BTW, could you add check, that user, which tried to sign-in is not banned?